PCM-2807 strataService in common missing articles.get

engineersamuel opened the pull request:
Adding articles.get so we can fetch the full html content for articles, PCM-2807

@vrathee @gandhikeyur @jtrantin  please review and push the new release to bower.

jtrantin commented on app/common/services/strataService.js:
```diff
@@ -197,6 +197,25 @@ angular.module('RedhatAccess.common').factory('strataService', [
+                get: function (uri) {
+                    var deferred = $q.defer();
+                    var splitUri = uri.split('/');
+                    uri = splitUri[splitUri.length - 1];
```
Correct me if I am wrong, but this code seems to parse the article ID out of the URI. Why are you not passing the ID of the article instead? Seems like a hassle to pass the URI just to extract the ID out of it.

engineersamuel replied:
I literally copy and pasted from the service part fetching solutions.  We could definitely rewrite it to be more sane but I want going to spend too much time on it in hopes that we can get this functionality in uds.  I have a Jira open for getting the HTML in solutions and articles.  Feel free to reject this though and I'll rewrite it on Monday.

engineersamuel replied:
Oh and you can just pass the id, you don't have to pass the full uri

jtrantin replied:
Still, seems rather hackish for no good reason. I personally would rewrite this, but let's wait for opinions from others :).

vrathee commented on app/common/services/strataService.js:
```diff
@@ -197,6 +197,25 @@ angular.module('RedhatAccess.common').factory('strataService', [
+                    var deferred = $q.defer();
+                    var splitUri = uri.split('/');
+                    uri = splitUri[splitUri.length - 1];
+                    if (!ie8 && strataCache.get('article' + uri)) {
```
Since we don't support ie8 now.. we can get rid of it.

vrathee commented:
Apart from what @jtrantin said regarding using actual article ID if we have it and the ie8 part. Rest Seems good.

renujhamtani commented:
@engineersamuel  There are comments from developers. Are we solving this or should i merge this?

engineersamuel commented:
@renujhamtani not quite yet, I'm about to push a few changes, then we can merge.

engineersamuel commented:
@jtrantin ok, cleaned up the code and rewrote it not following the prior way.  Ultimately we probably need to rewrite all methods to clean them up.  That'll be for a later time though.

@renujhamtani we should be good to merge now.

jtrantin commented:
Looks great :)

vrathee commented:
Reviewed!
